core/server/worker/src/test/java/alluxio/worker/block/CacheRequestManagerTest.java

```diff
@@ -0,0 +1,169 @@
+    CacheRequest request = CacheRequest.newBuilder().setBlockId(blockId).setLength(blockLength)
+        .setOpenUfsBlockOptions(openUfsBlockOptions).setSourceHost(localWorkerHostname)
+        .setSourcePort(port).build();
+    BlockReader mockBlockReader = mock(BlockReader.class);
+    when(mBlockWorker.createUfsBlockReader(Sessions.ASYNC_CACHE_UFS_SESSION_ID, blockId, 0, false,
+        openUfsBlockOptions)).thenReturn(mockBlockReader);
+    mCacheRequestManager.submitRequest(request);
+    verify(mBlockWorker).createUfsBlockReader(Sessions.ASYNC_CACHE_UFS_SESSION_ID, blockId, 0,
+        false, openUfsBlockOptions);
+  }
+
+  @Test
+  public void submitRequestCacheBlockFromRemoteWorker() throws Exception {
+    String localWorkerHostname = NetworkAddressUtils.getLocalHostName(
+        (int) ServerConfiguration.getMs(PropertyKey.NETWORK_HOST_RESOLUTION_TIMEOUT_MS));
+    String fakeRemoteWorker = localWorkerHostname + "1";
+    int port = mRandom.nextInt(5000);
+    long blockId = mRandom.nextLong();
+    long blockLength = mRandom.nextInt(5000);
+    Protocol.OpenUfsBlockOptions options = Protocol.OpenUfsBlockOptions.newBuilder()
+        .setMaxUfsReadConcurrency(10).setUfsPath("/a").build();
+    CacheRequest request = CacheRequest.newBuilder().setBlockId(blockId).setLength(blockLength)
+        .setOpenUfsBlockOptions(options).setSourceHost(fakeRemoteWorker)
+        .setSourcePort(port).build();
+    setupMockReaderWriter(fakeRemoteWorker, port, blockId, blockLength, options);
+    mCacheRequestManager.submitRequest(request);
+    verify(mBlockWorker).createBlock(Sessions.ASYNC_CACHE_WORKER_SESSION_ID, blockId, 0, "",
+        blockLength);
+    verify(mBlockWorker).createBlockWriter(Sessions.ASYNC_CACHE_WORKER_SESSION_ID, blockId);
+    verify(mBlockWorker).commitBlock(Sessions.ASYNC_CACHE_WORKER_SESSION_ID, blockId, false);
+  }
+
+  @Test
+  public void submitAsyncRequestCacheBlockFromUfs() throws Exception {
+    String localWorkerHostname = NetworkAddressUtils.getLocalHostName(
+        (int) ServerConfiguration.getMs(PropertyKey.NETWORK_HOST_RESOLUTION_TIMEOUT_MS));
+    int port = mRandom.nextInt(5000);
+    long blockId = mRandom.nextLong();
+    long blockLength = mRandom.nextInt(5000);
+    Protocol.OpenUfsBlockOptions openUfsBlockOptions = Protocol.OpenUfsBlockOptions.newBuilder()
+        .setMaxUfsReadConcurrency(10).setUfsPath("/a").build();
+    CacheRequest request = CacheRequest.newBuilder().setBlockId(blockId).setLength(blockLength)
+        .setOpenUfsBlockOptions(openUfsBlockOptions).setSourceHost(localWorkerHostname)
+        .setSourcePort(port).setAsync(true).build();
+    BlockReader mockBlockReader = mock(BlockReader.class);
+    when(mBlockWorker.createUfsBlockReader(Sessions.ASYNC_CACHE_UFS_SESSION_ID, blockId, 0, false,
+        openUfsBlockOptions)).thenReturn(mockBlockReader);
+    mCacheRequestManager.submitRequest(request);
+    Thread.sleep(100);  //sleep since it's async
```

Comment:
`Thread.sleep` is not reliable to test async ops, check for `CommonUtils.waitFor` and its usage

Reply:
not sure how we can use waitFor here since it's using a lot of mock. Use timeout with a large number instead and it should offer similar functionality since it will immediately pass if it becomes true at any point